app/src/org/commcare/dalvik/application/CommCareApplication.java

```
@@ -356,6 +343,10 @@ public void initializeGlobalResources(CommCareApp app) {
     }
 
     public String getPhoneId() {
+        if (ContextCompat.checkSelfPermission(this, Manifest.permission.READ_PHONE_STATE) == PackageManager.PERMISSION_DENIED) {
+            return "000000000000000";

```

Default to fake imei if we don't have correct permissions